DfE.FindInformationAcademiesTrusts/Services/Export/Builders/OfstedDataBuilder.cs

```diff
@@ -0,0 +1,129 @@
+﻿using DfE.FindInformationAcademiesTrusts.Data;
+using DfE.FindInformationAcademiesTrusts.Data.Repositories.Academy;
+using DfE.FindInformationAcademiesTrusts.Data.Repositories.Trust;
+using DfE.FindInformationAcademiesTrusts.Extensions;
+
+namespace DfE.FindInformationAcademiesTrusts.Services.Export.Builders
+{
+    public class OfstedDataBuilder(string sheetName = "Ofsted") : ExportBuilder(sheetName)
+    {
+        public OfstedDataBuilder WriteTrustInformation(TrustSummary? trustSummary)
+        {
+            WriteTrustInformation<OfstedDataBuilder>(trustSummary);
+            return this;
+        }
+
+        public OfstedDataBuilder WriteHeaders()
+        {
+            var headers = new List<string>
+            {
+                ExportHelpers.SchoolName,
+                ExportHelpers.DateJoined,
+                "Current single headline grade",
+                ExportHelpers.BeforeOrAfterJoiningHeader,
+                "Date of Current Inspection",
+                "Previous single headline grade",
+                ExportHelpers.BeforeOrAfterJoiningHeader,
+                "Date of previous inspection",
+                "Quality of Education",
+                "Behaviour and Attitudes",
+                "Personal Development",
+                "Leadership and Management",
+                "Early Years Provision",
+                "Sixth Form Provision",
+                "Previous Quality of Education",
+                "Previous Behaviour and Attitudes",
+                "Previous Personal Development",
+                "Previous Leadership and Management",
+                "Previous Early Years Provision",
+                "Previous Sixth Form Provision",
+                "Effective Safeguarding",
+                "Category of Concern"
+            };
+
+            WriteHeaders(headers);
+
+            return this;
+        }
+
+        public OfstedDataBuilder WriteRows(AcademyDetails[] academies, AcademyOfsted[] academiesOfstedRatings)
+        {
+            foreach (var details in academies)
+            {
+                var ofstedData = academiesOfstedRatings.SingleOrDefault(x => x.Urn == details.Urn);
+
+                GenerateOfstedRow(details, ofstedData);
+                AddRow();
+            }
+
+            return this;
+        }
+
+        private void GenerateOfstedRow(AcademyDetails academy, AcademyOfsted? ofstedData)
+        {
+            var previousRating = ofstedData?.PreviousOfstedRating ?? OfstedRating.NotInspected;
+            var currentRating = ofstedData?.CurrentOfstedRating ?? OfstedRating.NotInspected;
+
+            // School Name
+            SetTextCell(CurrentRow, 1, academy.EstablishmentName ?? string.Empty);
```

Comment:
It would be good to remove these magic column numbers, perhaps by moving the columns enum out of the test class?